File: go/solutions/1512-number-of-good-pairs.go
package solutions

func NumIdenticalPairs(nums []int) int {
	set := make(map[int]int)
	for i := range nums {
		set[nums[i]]++
	}

	count := 0
	for _, val := range set {
		count += (val * (val - 1)) / 2
	}

	return count
}

File: js/1512-number-of-good-pairs.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var numIdenticalPairs = function (nums) {
  var set = {},
    count = 0;
  nums.forEach((num) => (set[num] = (set[num] || 0) + 1));
  Object.values(set).forEach(
    (val) => (count += Math.floor((val * (val - 1)) / 2))
  );
  return count;
};

module.exports = { numIdenticalPairs };
